### PR TITLE
Update 2.3_autograd.ipynb 

### PR DESCRIPTION
**Pull request by haoma7**
Change `tensor.data` to `tensor.detach()` due to
https://github.com/pytorch/pytorch/issues/6990#issuecomment-384680164
`tensor.detach()` is more robust than `tensor.data`.

### CODE REVIEW

**Review by cyrus303 — APPROVED** [review]
looks good!

**Review by ghost — APPROVED** [review]
